# Clinical Computational Genomics

<p align="justify">The Clinical Computational Genomics course, as outlined in this repository, provides a comprehensive introduction to the algorithmic and statistical methods used to analyze large-scale genomic data with a focus on medical applications (Infectious Diseases and Human Genetic Diseases). The curriculum covers foundational bioinformatics topics such as sequence alignment CLustal, BWA, Bowtie (including Smith-Waterman and Needleman-Wunsch algorithms), BLAST and BLAT for gene prediction, gene annotation, and phylogenetic analysis. Additionally, the course delves into modern clinical workflows, including high-throughput sequencing data processing (FASTQC, Trimmomatic, Trimgalore...), variant calling (BCFTools and GATK), and the interpretation of genetic data for disease diagnosis and precision medicine. By combining theoretical lectures with practical computational exercises, the course prepares students to translate complex biological sequences into actionable clinical insights.</p>

## Part I: Background Preparation
### Lecture 0: Introduction to Next Generation Sequencing (NGS) and Long Read Sequencing in Clinical Genomics
- PDF
- Database
### Lecture 2: Introduction to Linux Operation System, Google Drive & Colab, Github and Linux Command Lines
- PDF
- Lab GGColab
- Homework GGColab
### Lecture 3: Introduction to Programming Language R & Data Visualization with GGPlot
- PDF
- Lab GGColab
- Homework GGColab
##
## Part II: Bioinformatic Analysis for Whole Genome Sequencing (WGS) of infection pathogens (Bacteria and Fungi)
### Lecture 4: Detecting and Characterizing infection pathogens using Whole Genome Sequencing (WGS) in Clinical Setting
- PDF
- Lab GGColab: Whole-Genome Sequencing (WGS) Basic Analysis: From Data Quality Control, Mapping, Variant Calling to Clinical Repor
- Homework GGColab
### Lecture 5: Bacterial WGS Essentials — MLST Typing and Antimicrobial Resistance (AMR) Gene Profiling
- PDF
- Lab GGColab:  MLST typing
- Homework GGColab
### Lecture 6: Fungal WGS Essentials — Targeted Resistance Markers from Whole-Genome Data
- PDF
- Lab GGColab: Assembly genome, QC genome
- Homework GGColab
##
## Part III: Bioinformatic Analysis for Human Microbiome
### Lecture 7: Introduction to Microbiome
- PDF
- Lab GGColab: Microbiome data exploration
- Homework GGColab
### Lecture 8: Sequencing strategies for Microbiome Analysis: Amplicon Sequencing and Shotgun Metagenomic
- PDF
- Lab GGColab: Amplicon Sequencing and Shotgun Metagenomic pipelines
- Homework GGColab
### Lecture 9: Clinical Application of Microbiome Analysis
- PDF
- Lab GGColab: Applying Amplicon Sequencing for Diagnosis
- Homework GGColab
##
## Part IV: Bioinformatic Analysis for Human Genomics
### Lecture 10: Introduction to Targeted Sequencing for Germline Inherited Disease Detection
- PDF
- Lab GGColab: Raw Sequence Processing & Alignment
- Homework GGColab
### Lecture 11: Introduction to Low-Pass (Coverage) Whole Genome Sequencing (WGS) for Non-Invasive Prenatal Testing (NIPT) & Pre-implantation Genetic Testing for Aneuploidy (PGT-A) 
- PDF
- Lab GGColab: Variant Calling and Filtering
- Homework GGColab
### Lecture 12: Introduction to Targeted Sequencing for Somatic Mutation Detection in Cancer using Tissue & Liquid Biopsy
- PDF
- Lab GGColab: Variant Annotation & Annotation Databases
- Homework GGColab
##
